### PR TITLE
adding GetTemplateByName method to the Client (#115)

### PR DESCRIPTION
**Pull request by eslutsky**
## Please describe the change you are making

This PR adds GetTemplateByName Method to the Template Client.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

File: template_get_by_name.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+								EBug,
+								"failed to convert Template %s",
+								templateName,
+							)
+						}
+
+					}
+				}
+			}
+			return nil
```
Comment (janosdebugs) [inline template_get_by_name.go:36]: The case where no template is found is not handled here.

**Review by janosdebugs — CHANGES_REQUESTED** [review]

File: template_get_by_name.go (hunk 1 above)
Comment (eslutsky) [reply template_get_by_name.go:44]: fixed

File: template_get_by_name.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		retries,
+		func() error {
+			response, err := o.conn.SystemService().TemplatesService().List().Search("name=" + templateName).Send()
+			if err != nil {
+				return err
+			}
+			for _, sdkObject := range response.MustTemplates().Slice() {
+				if mTemplate, ok := sdkObject.Name(); ok {
+					if templateName == mTemplate {
+						result, err = convertSDKTemplate(sdkObject, o)
```
Comment (janosdebugs) [inline template_get_by_name.go:21]: Why not return here immediately if there is a result?

File: template_get_by_name.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+								EBug,
+								"failed to convert Template %s",
+								templateName,
+							)
+						}
+
+					}
+				}
+			}
+			if result == nil {
```
Comment (janosdebugs) [inline template_get_by_name.go:34]: This is not needed if you return in the loop.

**Review by janosdebugs — CHANGES_REQUESTED** [review]

File: template_get_by_name.go (hunk 2 above)
Comment (eslutsky) [reply template_get_by_name.go:21]: done

File: template_get_by_name.go (hunk 3 above)
Comment (eslutsky) [reply template_get_by_name.go:34]: done

**Review by janosdebugs — APPROVED** [review]

**Pull request merged**